src/homework10/hashmap/HashMap.java

```diff
@@ -0,0 +1,122 @@
+package homework10.hashmap;
+
+import java.util.NoSuchElementException;
+
+public class HashMap<K, V> {
+    private static final int INITIAL_SIZE = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static int size = 0;
+    private Entry<K, V>[] entries;
+
+    private static class Entry<K, V> {
+        private K key;
+        private V value;
+        private Entry<K, V> entryNext;
+
+        public Entry(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+
+        public K getKey() {
+            return key;
+        }
+
+        public V getValue() {
+            return value;
+        }
+
+        public void setValue(V value) {
```

Comment:
maybe better add setKey()?

Reply:
I didn't use setKey() method, but if is it convention - I add.